### PR TITLE
[NETBEANS-4029] Fix feature dialog

### PR DESCRIPTION
**Pull request by hectorespert**
Fix feature dialog: https://issues.apache.org/jira/browse/NETBEANS-4029

### CODE REVIEW

File: ergonomics/ide.ergonomics/src/org/netbeans/modules/ide/ergonomics/fod/ConfigurationPanel.java [hunk 1]
```diff
@@ -190,6 +180,40 @@ public void setInfo(FeatureInfo info, String displayName, Collection<UpdateEleme
...
+        return descriptionsList;
+    } 
+    
+    /**
+     * Generate download message from extra modules texts
+     * @param extraModulesTexts
+     * @return String Text to set in download label
+     */
+    protected String generateDownloadMessageFromExtraModulesTexts(List<String> extraModulesTexts) {
+        StringBuilder sbDownload = new StringBuilder();
```
Comment (sdedic) [inline ergonomics/ide.ergonomics/src/org/netbeans/modules/ide/ergonomics/fod/ConfigurationPanel.java:204]: A suggestion: these two functions (`collectExtraModulesTextsFromFeatures` and `generateDownloadMessageFromExtraModulesTexts`) are called in a sequence: you could merge it, producing the String for the label - and you could eventually print the *required* texts in Bold into the string ?

File: ergonomics/ide.ergonomics/src/org/netbeans/modules/ide/ergonomics/fod/ConfigurationPanel.java [hunk 2]
```diff
@@ -190,6 +180,40 @@ public void setInfo(FeatureInfo info, String displayName, Collection<UpdateEleme
...
     }
+    
+    /**
+     * Collect extra modules texts
+     */
+    protected List<String> collectExtraModulesTextsFromFeatures(Collection<FeatureInfo> features, boolean required) {
+        List<String> descriptionsList = new ArrayList<>();
+        for (FeatureInfo fi : features) {
+            String s = required ? fi.getExtraModulesRequiredText(): fi.getExtraModulesRecommendedText();
+            if (!descriptionsList.contains(s)) {
```
[the remark this answers is not in the document]
Comment (hectorespert) [reply ergonomics/ide.ergonomics/src/org/netbeans/modules/ide/ergonomics/fod/ConfigurationPanel.java]: No, I added the null check

File: ergonomics/ide.ergonomics/src/org/netbeans/modules/ide/ergonomics/fod/ConfigurationPanel.java (hunk 1 above)
Comment (hectorespert) [reply ergonomics/ide.ergonomics/src/org/netbeans/modules/ide/ergonomics/fod/ConfigurationPanel.java:206]: I could merge both functions, but I created these to allow create tests for this part of the code. 
Is not the best solution but is too hard to test the ConfigurationPanel class due his dependencies.

Comment (sdedic) [reply ergonomics/ide.ergonomics/src/org/netbeans/modules/ide/ergonomics/fod/ConfigurationPanel.java:206]: OK, it was just an idea; let's see how the entire message list will be readable

File: ergonomics/ide.ergonomics/src/org/netbeans/modules/ide/ergonomics/fod/ConfigurationPanel.java [hunk 3]
```diff
@@ -188,7 +190,7 @@ public void setInfo(FeatureInfo info, String displayName, Collection<UpdateEleme
         List<String> descriptionsList = new ArrayList<>();
         for (FeatureInfo fi : features) {
             String s = required ? fi.getExtraModulesRequiredText(): fi.getExtraModulesRecommendedText();
-            if (!descriptionsList.contains(s)) {
+            if (nonNull(s) && !descriptionsList.contains(s)) {
```
Comment (sdedic) [inline ergonomics/ide.ergonomics/src/org/netbeans/modules/ide/ergonomics/fod/ConfigurationPanel.java:193]: While the check does its job isn't it a little overabstracting a simple `s != null` :-)  ?
if there is an explicit `import static` for the `nonNull`, the preceding `import j.u.Objects` is not needed.

Comment (hectorespert) [reply ergonomics/ide.ergonomics/src/org/netbeans/modules/ide/ergonomics/fod/ConfigurationPanel.java:192]: I removed unused import. I forgot to remove it.

Comment (ebarboni) [reply ergonomics/ide.ergonomics/src/org/netbeans/modules/ide/ergonomics/fod/ConfigurationPanel.java:192]: is this resolved ? can we merge ?

Comment (hectorespert) [reply ergonomics/ide.ergonomics/src/org/netbeans/modules/ide/ergonomics/fod/ConfigurationPanel.java:192]: It is resolved

**Pull request merged**